mpa/cls/inferrer.py

```diff
@@ -90,26 +97,21 @@ def _infer(self, cfg):
                 data_info['soft_label'] = {task: value[i] for task, value in old_prob.items()}
             outputs = data_infos
         else:
-            outputs = self.single_gpu_test(model, data_loader)
-
+            with FeatureVectorHook(model.module.backbone) if dump_features else nullcontext() as fhook:
+              with SaliencyMapHook(model.module.backbone) if dump_saliency_map else nullcontext() as shook:
+                  for data in data_loader:
+                      with torch.no_grad():
+                          result = model(return_loss=False, **data)
+                      eval_predictions.extend(result)
+                  feature_vectors = fhook.records if dump_features else [None] * len(self.dataset)
+                  saliency_maps = shook.records if dump_saliency_map else [None] * len(self.dataset)
+
+        assert len(eval_predictions) == len(feature_vectors) == len(saliency_maps), \
+                'Number of elements should be the same, however, number of outputs are ' \
+                f"{len(eval_predictions)}, {len(feature_vectors)}, and {len(saliency_maps)}"
+        outputs = dict(
+            eval_predictions=eval_predictions,
+            feature_vectors=feature_vectors,
+            saliency_maps=saliency_maps
+        )
```

Comment:
Need to insert `model.eval()`

Reply:
model.eval() has been reflected in the latest develop branch, so you can rebase it.